test/runtests.cmd

```diff
@@ -474,6 +475,7 @@ goto :main
 
   echo ^>^> %*
   cmd /s /c "%*"
+  if %ERRORLEVEL% NEQ 0 exit /b %ERRORLEVEL%
 

```

Comment:
This line seems at odds with the addition on line 448 above.  This will prevent that line from executing, no?

More importantly though, `:do` is meant to just print out its command before executing it.  It should not be taking control of control flow.